lib/Form/Processor/RoleFormProcessor.php

```diff
@@ -38,20 +38,22 @@ public static function getSubscribedEvents()
     public function processDefaultAction(FormActionEvent $event)
     {
         // Don't update anything if we just want to cancel the draft.
-        if ($event->getClickedButton() === 'removeDraft') {
+        // Don't do the default action when saving is specified, as it's the same action and shouldn't be done twice.
+        if (in_array($event->getClickedButton(), ['removeDraft', 'saveRole'])) {
             return;
         }
 
-        /** @var \EzSystems\RepositoryForms\Data\Role\RoleData $roleData */
```

Comment:
Naming nitpick, but I'd try to replace this typehinting, done 3 times in the class, with a better named method with proper phpdoc (`$this->getRoleData()` ?).

What would make it the cleanest would be a decoration of the event we get, with methods that know about the underlying data (`$event->getRoleDraft()`, for instance). But not now :-)